src/Collectors/ClassConstFetchCollector.php

```diff
@@ -60,7 +60,7 @@ public function processNode(Node $node, Scope $scope): ?array
             if ($classReflection->hasConstant($constantName)) {
                 $constantReflection = $classReflection->getConstant($constantName);
                 $declaringClass = $constantReflection->getDeclaringClass();
-                if ($declaringClass->getFileName() !== $classReflection->getFileName()) {
+                if ($declaringClass->getName() !== $classReflection->getName()) {
```

Comment:
I think its a typo, that this check uses the file-name instead of the class-name.

in files in which several classes are used in a single file, the previous check would not work